app/src/main/java/com/github/sdp_begreen/begreen/fragments/MapFragment.kt

```diff
@@ -151,51 +176,48 @@ class MapFragment : Fragment() {
     }
 
     /**
-     * Helper functions that displays bins markers on the map
+     * Helper functions that displays bins markers on the map, and sets up click events
+     * Only called once when the view is created
      */
-    private fun displayBinsMarkers(bins: Set<Bin>) {
-
-        // Clear old bins so we don't display removed or duplicates bins
-        map.clear()
-
-        for (bin in bins) {
-
-            val location = LatLng(bin.lat, bin.long)
-
-            val marker = map.addMarker(
-                MarkerOptions()
-                    .position(location)
-                    .title(bin.type.toString())
-                    .icon(BitmapDescriptorFactory.defaultMarker(bin.type.markerColor))
-            )
-
-            marker!!.tag = bin.id
-        }
+    private fun displayBinsMarkers(bins: Set<Bin>) = bins.forEach{ addMarker(it) }
 
-        // Setup the markers click listener action
-        setupMarkersClick()
-    }
 
     /**
-     * Helper function that setups a marker click listener action
+     * Helper function that setups a marker's info window click listener action
      */
-    private fun setupMarkersClick() {
+    private fun setupInfoWindowClick() {
```

Comment:
That's a great idea! I prefer to spend time on my second task for the moment since I think it's more important than this modification, but we can definitely plan to make this modification later.